### ISSUE TITLE
The price of recipe and amount to pay should declare in double.

### ISSUE THREAD

**Issue by NorSamsudin** [smashita/CPT341SDA#45]
`<image.src=priceOfRecipe>`
![priceofrecipe](https://user-images.githubusercontent.com/43567339/47255966-f97c4580-d4aa-11e8-96d0-2ac75b448035.PNG)

From this screenshoot image shown that the recipe was not added because there is error at the price.

`<image.src=AmountToPay>`
![amounttopay](https://user-images.githubusercontent.com/43567339/47256108-e5394800-d4ac-11e8-80aa-0e5b98ed774b.PNG)

Also same for this screenshoot image shown that the coffe cannot make because there is error at the amount to pay.

So, as we know that money not always in integer. I suggest to change the declaration of recipe price and amount to pay into double.

**Comment by surayamahtab** [comment]
That is right. Price of beverages need to allowed to be in cents either for future price increment.

**Comment by nafre** [comment]
From what I remembered, price should not be declared as double or float because it is impossible to represent 0.1 as a float or double exactly. 

> Representing money as a double or float will probably look good at first as the software rounds off the tiny errors, but as you perform more additions and subtractions on inexact numbers, errors will compound and you'll end up with values that are visibly not accurate. This makes floats and doubles inadequate for dealing with money, where perfect accuracy for multiples of base 10 powers is required.
[Source](https://stackoverflow.com/a/3730040)

The best solution here is to either change it to BigDecimal class or remain as integer.

**Comment by nafre** [comment]
From what I remembered, price should not be declared as double or float because it is impossible to represent 0.1 as a float or double exactly.

**Comment by chuazhe** [comment]
> From what I remembered, price should not be declared as double or float because it is impossible to represent 0.1 as a float or double exactly.
> 
> Edit:
> 
> > Representing money as a double or float will probably look good at first as the software rounds off the tiny errors, but as you perform more additions and subtractions on inexact numbers, errors will compound and you'll end up with values that are visibly not accurate. This makes floats and doubles inadequate for dealing with money, where perfect accuracy for multiples of base 10 powers is required.
> > [Source](https://stackoverflow.com/a/3730040)
> 
> The best solution here is to either change it to BigDecimal class or remain as integer.

agree, only BigDecimal can give the the best support when it comes to precision of the money

**Comment by alexleong4438** [comment]
> From what I remembered, price should not be declared as double or float because it is impossible to represent 0.1 as a float or double exactly.
> 
> Edit:
> 
> > Representing money as a double or float will probably look good at first as the software rounds off the tiny errors, but as you perform more additions and subtractions on inexact numbers, errors will compound and you'll end up with values that are visibly not accurate. This makes floats and doubles inadequate for dealing with money, where perfect accuracy for multiples of base 10 powers is required.
> > [Source](https://stackoverflow.com/a/3730040)
> 
> The best solution here is to either change it to BigDecimal class or remain as integer.

I'm agree with you,  BigDecimal is a good way to deal with money. However,disadvantage of BigDecimal is that it's slower compare with double, and it's a bit more difficult to program algorithms that way (due to + - * and / not being overloaded).
